### PR TITLE
Unit test with testcontainers 

### PR DESCRIPTION
**Pull request by hoangthanh28**
Hi Erik.
I plan to contribute to **grate** because I love it. It's a great tool, please consider my PR with the following changes:
- [x] Refactoring the current unit test  with [testcontainers](https://dotnet.testcontainers.org/)
- [x] Migrate the current NUnit to xUnit (I believe xUnit will be better for us in the long run)
- [x] Speedup the SqlServer Unit test, the duration is reduced from 21mins to 2mins
- [x] Adjust the code format using *dotnet format* (I'm using Debian 12, vs code and C# DevKit). Please let me know what the best option is for us to follow in terms of code formatting.  Feel free to prune/reformat the code if needed. Sorry for the inconsistent style with yours.

### CODE REVIEW

**Comment by erikbra** [comment]
Hi, @hoangthanh28. Thanks a lot for your PR! Great that you love great (pun intended), and even better that you would like to contribute! This is what open source should be, collaboration! :)

I originally used NUnit for grate, partly because it was used in RoundhousE, and partly because the "run this thing _once_ before all tests" was easier in NUnit, at least I thought so at the time. So I used that for setting up the containers before running the tests. But I like xUnit better personally as well.

Your PR is quite big, so I'll have to take a little while looking into it, but initially I'm positive to the initiatives!

**Comment by hoangthanh28** [comment]
Thank you Erik for your consideration. Btw, I'm a 5-year RoundHousE user, as well as a big fan of database first, changing database is crucial, so we need to script it down and put into the version control system. :)

File: grate/Configuration/IFoldersConfiguration.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 namespace grate.Configuration;
 
-public interface IFoldersConfiguration: IDictionary<string, MigrationsFolder?>
+public interface IFoldersConfiguration : IDictionary<string, MigrationsFolder?>
```
Comment (erikbra) [inline grate/Configuration/IFoldersConfiguration.cs:5]: I'm not used to having i space before the colon in inheritance clauses, but I see that the standard for this is `true`, and I don't want to have a lot of special naming rules (as RoundhousE had a _lot_ of, the naming style there is quite exotic 😆 )

But, Let's go with this, as that is standard :)

csharp_space_before_colon_in_inheritance_clause

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/csharp-formatting-options#csharp_space_before_colon_in_inheritance_clause

**Comment by erikbra** [comment]
Thanks again, @hoangthanh28 - I have looked through it, and it looks very clean and good quality code, I like it!
In general, I would prefer having only one type of change in each PR, it's easier to see what is changed (e.g. changing to TestContainers, _or_ changing to xUnit), but I understand that they might walk a bit hand-in-hand, as the dependency injection might have changed.

I'm going to merge this as-is. I have a few suggestions for improvements, but I'll just make a separate PR myself into main for them.

**Pull request merged**